engine/src/main/java/org/hibernate/validator/metadata/descriptor/HibernateConstraintDescriptor.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.metadata.descriptor;
+
+import java.lang.annotation.Annotation;
+
+import javax.validation.metadata.ConstraintDescriptor;
+
+/**
+ * Extension to {@code ConstraintDescriptor} which provides functionality
+ * specific to Hibernate Validator.
+ *
+ * @author Guillaume Smet
+ * @since 6.0
+ */
+public interface HibernateConstraintDescriptor<T extends Annotation> extends ConstraintDescriptor<T> {
```

Comment:
As discussed on HipChat I'm not in favor of adding it prematurely. People will only will wonder what the empty interface is about. For the `unwrap()` implementation, we can simply raise an exception at this point, as no type to unwrap to currently is supported. The alternative is to allow to unwrap into `ConstraintDescriptor` and super-types. I don't mind either way.